src/benchmarks/pmem_memset.cpp

```diff
@@ -298,20 +327,39 @@ memset_init(struct benchmark *bench, struct benchmark_args *args)
 		goto err_free_offsets;
 	}
 
-	if (mb->pargs->memset)
-		mb->func_op = (mb->pargs->persist) ? libc_memset_persist
-						   : libc_memset;
-	else
+	if (mb->pargs->memset) {
+		if (mb->pargs->persist && mb->pargs->msync) {
+			fprintf(stderr, "Invalid benchmark parameters: "
+					"persist and msync cannot be specified "
+					"together\n");
+			ret = -1;
+			goto err_free_offsets;
+		}
+
+		if (mb->pargs->persist) {
+			mb->func_op = libc_memset_persist;
+		} else if (mb->pargs->msync) {
+			mb->func_op = libc_memset_msync;
+			warmup_func = warmup_msync;
+		} else {
+			mb->func_op = libc_memset;
+		}
+	} else {
 		mb->func_op = (mb->pargs->persist) ? libpmem_memset_persist
 						   : libpmem_memset_nodrain;
+	}
 
 	if (!mb->pargs->no_warmup) {
-		do_warmup(mb, args->n_threads * args->n_ops_per_thread);
+		ret = warmup_func(mb);
+		if (ret) {
+			fprintf(stderr, "Pool warmup failed\n");
```

Comment:
Warmup can fail??
Because of msync, right?
then:
`perror("Pool warmup failed");